### DISCUSSION TITLE
Password Policy for K3os

### DISCUSSION THREAD

**Discussion by kuddusrhce** [rancher/k3os#797] [Q&A]
Hello Team,

We are planning to implement a password policy on K3OS for audit purposes. How to achieve this in k3os so that we can implement password policies for custom users in K3OS?

**Comment by dweomer** [comment]
k3OS comes with one user configured besides `root` and that is `rancher`. The password for the `rancher` user is sourced from the `config.yaml` and set to whatever is found there at every boot (`/etc/` is ephemeral). Additionally, this can be overridden by the `k3so.password` kernel parameter.

If you wish to manage other users to the system you will have to cobble that together on your own which will not be fun.

**Comment by kuddusrhce** [comment]
Thank you for your quick reply. If we want to apply password policies such as

Password Max days.
Password Min days.
Password warning days.
Password history or Deny Re-Used Passwords.
Password minimum length.
Minimum upper case characters.
Minimum lower case characters.
Minimum digits in password.

How to apply password policies for root and rancher users? Also, how can we create multiple users and store them permanently as /etc is the ephemeral file system.
